plugins/flytekit-pydantic/tests/test_type_transformer.py

```diff
@@ -0,0 +1,203 @@
+        (ConfigRequired, {"model_config": {"foo": TrainConfig(loss="mse")}}),
+        (ConfigWithFlyteFiles, {"flytefiles": ["s3://foo/bar"]}),
+        (ConfigWithFlyteDirs, {"flytedirs": ["s3://foo/bar"]}),
+        (ConfigWithPandasDataFrame, {"df": pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})}),
+    ],
+)
+def test_pass_to_workflow(config_type: Type, kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = config_type(**kwargs)
+
+    @flytekit.task
+    def train(cfg: config_type) -> config_type:
+        return cfg
+
+    @flytekit.workflow
+    def wf(cfg: config_type) -> config_type:
+        return train(cfg=cfg)
+
+    returned_cfg = wf(cfg=cfg)
+
+    # assert returned_cfg == cfg
+    # TODO these assertions are not valid for all types
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"flytefiles": ["tests/folder/test_file1.txt", "tests/folder/test_file2.txt"]},
+    ],
+)
+def test_flytefiles_in_wf(kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = ConfigWithFlyteFiles(**kwargs)
+
+    @flytekit.task
+    def read(cfg: ConfigWithFlyteFiles) -> str:
+        with open(cfg.flytefiles[0], "r") as f:
+            return f.read()
+
+    @flytekit.workflow
+    def wf(cfg: ConfigWithFlyteFiles) -> str:
+        return read(cfg=cfg)  # type: ignore
+
+    string = wf(cfg=cfg)
+    assert string in {"foo", "bar"}  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "kwargs",
+    [
+        {"flytedirs": ["tests/folder/"]},
+    ],
+)
+def test_flytedirs_in_wf(kwargs: Dict[str, Any]):
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg = ConfigWithFlyteDirs(**kwargs)
+
+    @flytekit.task
+    def listdir(cfg: ConfigWithFlyteDirs) -> List[str]:
+        return os.listdir(cfg.flytedirs[0])
+
+    @flytekit.workflow
+    def wf(cfg: ConfigWithFlyteDirs) -> List[str]:
+        return listdir(cfg=cfg)  # type: ignore
+
+    dirs = wf(cfg=cfg)
+    assert len(dirs) == 2  # type: ignore
+
+
+def test_double_config_in_wf():
+    """Test passing a BaseModel instance to a workflow works."""
+    cfg1 = TrainConfig(batch_size=13)
+    cfg2 = TrainConfig(batch_size=31)
+
+    @flytekit.task
+    def are_different(cfg1: TrainConfig, cfg2: TrainConfig) -> bool:
+        return cfg1 != cfg2
+
+    @flytekit.workflow
+    def wf(cfg1: TrainConfig, cfg2: TrainConfig) -> bool:
+        return are_different(cfg1=cfg1, cfg2=cfg2)  # type: ignore
+
+    assert wf(cfg1=cfg1, cfg2=cfg2), wf(cfg1=cfg1, cfg2=cfg2)  # type: ignore
+
+
+# TODO: //Arthur to Fabio this was differente before but now im unsure what the test is doing
+# previously a pattern match error was checked that its raised, but isnt it OK that the ChildConfig
+# is passed since its a subclass of Config?
+# I modified the test to work the other way around, but im not sure if this is what you intended
```

Comment:
I see. 

To support the `schema()` method, we would need to define a schema for each of the flytetypes. 
Think it would require some design to make that generic across python versions and that leaving this one to the end user is likely to be OK.

What do you think?

Reply:
Notes from huddle:

We cannot use `.schema()` unless with use e.g. `TypeVar("pd.Dataframe")` as typehint  instead of `pd.Dataframe` but if we do that the validators are not invoked.

This test can be removed then.